lib/customerio/base_client.rb

```diff
@@ -0,0 +1,76 @@
+require 'net/http'
+require 'multi_json'
+
+module Customerio
+  DEFAULT_TIMEOUT  = 10
+
+  class BaseClient
+    def initialize(auth, options = {})
+      @auth = auth
+      @timeout = options[:timeout] || DEFAULT_TIMEOUT
+
+      @json = options.has_key?(:json) ? options[:json] : true
+      @base_uri = options[:base_uri] || DEFAULT_BASE_URI
```

Comment:
whats `DEFAULT_BASE_URI` in this case? feels like maybe it should be an error if missing